ext/src/ruby_api/func_type.rs

```diff
@@ -26,14 +29,32 @@ impl FuncType {
 }
 
 impl FuncType {
```

Comment:
I know this is more of a design comment, than a documentation comment, but does it make sense to expose this as a class in Ruby?

I had expected these to be part of the `Func` class, e.g.

```ruby

my_func = Wasmtime::Func.new(my_store, params: [:i32, :f64], results: [:v128]) do
  # …
end

# and then if introspection is needed, something like this:
my_func.params #=> [:i32, :f64]
my_func.results #=> [:v128]

```

`FuncType` could still be an internal class if needed, but I wouldn't expose it as a part of the public API.

Reply:
>I know this is more of a design comment (and this is a docs PR)

At this stage, those are the best kind of comments. Don't hold them back!

I agree that it feels redundant in Ruby. The reason I went with it is for consistency with Wasmtime: Rust, C and Python have that representation (but not .NET). There's value in being close to Wasmtime: similar experience across bindings, the canonical Wasmtime doc is more relevant, easier to stay in sync with Wasmtime, etc.

For this particular case though, does it matter? Probably not 🤔.

Reply:
I see the reason for it, even though I would lean in the other direction (tune the SDKs/bindings to the host language, i.e. make the Ruby bindings feel more "like ruby").

Maybe a private class would be a good middle ground?

You'd still get mostly 1:1 with Rust, C, Go and Python (from an implementation perspective).

But API-wise, you'd only expose these via `Func` (as in the example above).

If you'd later decide to expose `FuncType` as a public class, the old methods would just be syntactic sugar over using FuncType directly.

***

Somewhat related, I'd do the same thing with `MemoryType` (make it an internal class, expose relevant API directly on `Memory`.